lib/core/include/irods/configuration_hook_manager.hpp

```diff
@@ -0,0 +1,117 @@
+#ifndef IRODS_CONFIGURATION_HOOK_MANAGER_HPP
```

Comment:
This library is not actually specific to just configuration. We need to give this a more general name.

Discush!